src/MomentoCacheClient.php

```
@@ -1490,7 +1562,16 @@ public function setex(string $key, int $expire, mixed $value)
      */
     public function setnx(string $key, mixed $value): Redis|bool
     {
-        throw MomentoToPhpRedisExceptionMapper::createCommandNotImplementedException(__FUNCTION__);
+        $result = $this->client->setIfAbsent($this->cacheName, $key, $value);
+        if ($result->asStored()) {
+            return true;
+        } else if ($result->asNotStored()) {
+            return false;
+        } else {
+            logger()->error('Unexpected result from setnx', ['result' => $result]);
```

We should throw an exception on an error.